memory/combat_manager.py

```diff
@@ -0,0 +1,150 @@
+                    self.fields_base = self.memory.get_class_fields_base(singleton_ptr)
+                    self.current_encounter_base = self.memory.get_field(
+                        self.fields_base, "currentEncounter"
+                    )
+
+                # Update fields
+                self._read_players()
+                self._read_enemies()
+            else:
+                self.__init__()
+
+        except Exception:
+            return
+
+    def _read_players(self):
+        if self.memory.ready_for_updates():
+            player_panels_list = self.memory.follow_pointer(
+                self.base, [self.current_encounter_base, 0xE0, 0x80, 0x40, 0x0]
+            )
+            # item is a list of pointers of size 0x08
+            items = self.memory.follow_pointer(
+                player_panels_list,
+                [0x10, 0x0],
+            )
+            players = []
+            if items:
+                count = self.memory.read_int(items + 0x18)
+                address = 0x20
+                for _x in range(count):
+                    item = self.memory.follow_pointer(items, [address, 0x0])
+                    if hex(item) != "0x0":
+                        definition_id = self.memory.read_longlong(item + 0x70)
+                        selected = self.memory.read_int(item + 0x78)
+
+                        hp_text_field = self.memory.follow_pointer(item, [0x28, 0x0])
+                        current_hp = self.memory.read_int(hp_text_field + 0x54)
+
+                        portrait = self.memory.follow_pointer(item, [0x68, 0x0])
+                        enabled = self.memory.read_bool(portrait + 0x20)
+
+                        mp_text_field = self.memory.follow_pointer(item, [0x30, 0x0])
+                        current_mp = self.memory.read_int(mp_text_field + 0x54)
+                        players.append(
+                            {
+                                "current_hp": current_hp,
+                                "current_mp": current_mp,
+                                "definition_id": definition_id,
+                                "selected": selected,
+                                "enabled": enabled,
+                            }
+                        )
+                    address += 0x8
+
+            self.players = players
+            return
+        self.players = []
+
+    def _read_enemies(self):
+        if self.memory.ready_for_updates():
+            enemy_targets = self.memory.follow_pointer(
+                self.base, [self.current_encounter_base, 0x130, 0x0]
+            )
+            # item is a list of pointers of size 0x08
+            items = self.memory.follow_pointer(
+                enemy_targets,
+                [0x10, 0x0],
+            )
+            enemies = []
+            if items:
+                count = self.memory.read_int(items + 0x18)
+                address = 0x20
+                for _x in range(count):
```

Comment:
Could do something here like:

```
base_address = 0x20
for i in range(count):
    address = base_address + count * 0x8
    ...
    (remove address += 0x8)
```

Is it better? Dunno.

Reply:
both are fine - one less assignment and thing to gc with my approach